### ISSUE TITLE
Is there any way to capture & print the error message coming from Github API?

### ISSUE THREAD

**Issue by PawelLipski** [PawelLipski/git-machete#332]
I've got 422 on an attempt to open PR with base being a branch that was present locally, but already deleted on a remote... but the only error message I've got was:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/git_machete-3.5.0-py3.8.egg/git_machete/github.py", line 145, in __fire_github_api_request
  File "/usr/lib64/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib64/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/usr/lib64/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/usr/lib64/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/usr/lib64/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib64/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 422: Unprocessable Entity

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/git-machete", line 4, in <module>
    __import__('pkg_resources').run_script('git-machete==3.5.0', 'git-machete')
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 666, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/lib/python3.8/site-packages/pkg_resources/__init__.py", line 1469, in run_script
    exec(script_code, namespace, namespace)
  File "/usr/local/lib/python3.8/site-packages/git_machete-3.5.0-py3.8.egg/EGG-INFO/scripts/git-machete", line 19, in <module>
  File "/usr/local/lib/python3.8/site-packages/git_machete-3.5.0-py3.8.egg/git_machete/cli.py", line 803, in main
  File "/usr/local/lib/python3.8/site-packages/git_machete-3.5.0-py3.8.egg/git_machete/cli.py", line 622, in launch
  File "/usr/local/lib/python3.8/site-packages/git_machete-3.5.0-py3.8.egg/git_machete/client.py", line 2001, in create_github_pr
  File "/usr/local/lib/python3.8/site-packages/git_machete-3.5.0-py3.8.egg/git_machete/github.py", line 184, in create_pull_request
  File "/usr/local/lib/python3.8/site-packages/git_machete-3.5.0-py3.8.egg/git_machete/github.py", line 150, in __fire_github_api_request
git_machete.exceptions.UnprocessableEntityHTTPError: Unprocessable Entity
Please open an issue regarding this topic under link: https://github.com/VirtusLab/git-machete/issues/new
```

This doesn't help in determining the actual cause of the problem :/ Same problem as with #177... we could theoretically `if`-out all the cases like that explicitly, but maybe better to print the error message from Github instead (if available)?

If not, let's open an issue for handling non-existent base branch, similar to #177.

Also, instead of printing an ugly stack trace (or even worse, two ugly stack traces), the exception should be captured at the very top and printed similarly to `MacheteException`

**Comment by rifel123** [comment]
Both built-in internet python libraries (`urllib`, `http` -> they are suggesting to use `requests` on their docs site :)) are raising  `HTTPError` in such a case  of status 422 and exact error body from github is lost. Only with `requests` i was able to extract desired data like: `'A pull request already exists for rifel123:1.'` which is more descriptive (but also pretty well hidden, under: `response.text.message.errors.message`).
At current shape of git-machete with built-ins libraries to use, all we can extract is the response error, which is in case of status 422 is always `Unprocessable Entity`.

**Comment by PawelLipski** [comment]
😢

**Comment by PawelLipski** [comment]
Maybe adding at least some external deps isn't such a bad idea after all... but that would require **very** thorough testing in _Dry-run deploy_ steps to be implemented first

**Comment by PawelLipski** [comment]
Anyway! as for now, within the scope of this task, pls open an issue analogous to #177 + improve the way the exception is displayed to the users, and let's call it day after that.

**Comment by PawelLipski** [comment]
@rifel123 hmmm actually... does this help in any way? https://stackoverflow.com/a/52383241/2750812

**Comment by rifel123** [comment]
> @rifel123 hmmm actually... does this help in any way? https://stackoverflow.com/a/52383241/2750812

Yes, it does.
Currently this form is used:
<img width="874" alt="Screenshot 2021-11-08 at 09 08 55" src="https://user-images.githubusercontent.com/73847352/140706080-5b366988-0424-40ed-b49c-9bc2cfbd8a0b.png">

**Comment by PawelLipski** [comment]
Gr8!

**Comment by rifel123** [comment]
closed with #336
